master/internal/resourcemanagers/agent/agent.go

```diff
@@ -105,6 +111,16 @@ func (a *agent) Receive(ctx *actor.Context) error {
 func (a *agent) receive(ctx *actor.Context, msg interface{}) error {
 	switch msg := msg.(type) {
 	case actor.PreStart:
+		if a.agentState != nil { // not nil agentState on PreStart means it's restored.
+			a.started = true
+			a.agentState.Handler = ctx.Self()
+			// Update maxZeroSlotContainers config setting.
+			a.agentState.maxZeroSlotContainers = a.maxZeroSlotContainers
+			a.socketDisconnected(ctx)
+			// TODO(ilia): Adding restored agent here will overcount AgentStarts by maximum
+			// agentReconnectWait if it never reconnects.
+			ctx.Tell(a.resourcePool, sproto.AddAgent{Agent: ctx.Self(), Label: a.agentState.Label})
+		}
```

Comment:
maybe put this in a method like "postRestore". idk.

Reply:
it's somewhat awkward in the current system. the agent is tracked when it's added/removed from the resource pool. it's fine, most of the time, but it doesn't take into account disconnects like this one. 
That is, even before this PR, when an agent disconnects, we'd put the agent into draining mode, but it won't be removed from the pool until agent actor goes away after agent_reconnect_wait_timeout, also effectively overcounting by 25 seconds.

If we stop counting the agents when they disconnect we'd be undercounting them more instead when they lose connection but still run.

so, it's not quite an exact science either way, and I don't see a good way to improve it for restore flow specifically...

Reply:
cc @gt2345